src/timeaxis/timegrid.jl

```diff
@@ -58,21 +58,34 @@ Base.eltype(::Type{<:TimeGrid{T}}) where T = T
 Base.length(tg::TimeGrid{T,P,:finite}) where {T,P} = tg.n
 Base.size(tg::TimeGrid{T,P,:finite}) where{T,P}    = tg.n
 
+###############################################################################
+#  Empty TimeGrid
+###############################################################################
+Base.isempty(tg::TimeGrid{T,P,:finite}) where {T,P} = iszero(tg.n)
+Base.isempty(tg::TimeGrid{T,P,:infinite}) where {T,P} = false
+Base.isfinite(tg::TimeGrid{T,P,:finite}) where {T,P} = true
+Base.isfinite(tg::TimeGrid{T,P,:infinite}) where {T,P} = false
 
 ###############################################################################
 #  Printing
 ###############################################################################
 
-Base.show(io::IO, tg::TimeGrid{T,P,:finite}) where {T,P} =
-    print(io, "$(tg.o) … $(tg[end]) / $(tg.p) ")
+function Base.show(io::IO, tg::TimeGrid{T,P,:finite}) where {T,P}
+    summary(io, tg)
```

Comment:
well, I won't invoke `summary` here, since this function will be used for `repr`.
```suggestion

```